src/eva/plotting/batch/base/diagnostics/contour_plot.py

```diff
@@ -0,0 +1,112 @@
+from eva.eva_path import return_eva_path
+from eva.utilities.config import get
+from eva.utilities.utils import get_schema, update_object, slice_var_from_str
+import numpy as np
+
+from abc import ABC, abstractmethod
+
+# --------------------------------------------------------------------------------------------------
+
+
+class ContourPlot(ABC):
+
+    """Base class for creating Contour plots."""
+
+    def __init__(self, config, logger, dataobj):
+
+        """
+        Creates a Contour plot abstract class based on the provided configuration.
+
+        Args:
+            config (dict): A dictionary containing the configuration for the contour plot on a map.
+            logger (Logger): An instance of the logger for logging messages.
+            dataobj: An instance of the data object containing input data.
+
+
+        Example:
+
+            ::
+
+                    config = {
+                        "x": {"variable": "collection::group::variable"},
+                        "y": {"variable": "collection::group::variable"},
+                        "z": {"variable": "collection::group::variable"},
+                        "plot_property": "property_value",
+                        "plot_option": "option_value",
+                        "schema": "path_to_schema_file.yaml"
+                    }
+                    logger = Logger()
+                    scatter_plot = Scatter(config, logger, None)
```

Comment:
I think this is a cut & paste legacy.  This should be a contour plot example, right?